x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/operator/OperatorTestCase.java

```diff
@@ -80,15 +83,26 @@ public final void testSimpleLargeInput() {
      * in a sane way.
      */
     public final void testSimpleCircuitBreaking() {
-        BigArrays bigArrays = new MockBigArrays(PageCacheRecycler.NON_RECYCLING_INSTANCE, smallEnoughToCircuitBreak());
-        CircuitBreaker breaker = bigArrays.breakerService().getBreaker(CircuitBreaker.REQUEST);
-        BlockFactory blockFactory = BlockFactory.getInstance(breaker, bigArrays);
-        Exception e = expectThrows(
-            CircuitBreakingException.class,
-            () -> assertSimple(new DriverContext(bigArrays, blockFactory), between(1_000, 10_000))
-        );
-        assertThat(e.getMessage(), equalTo(MockBigArrays.ERROR_MESSAGE));
-        assertThat(bigArrays.breakerService().getBreaker(CircuitBreaker.REQUEST).getUsed(), equalTo(0L));
+        /*
+         * We build two CircuitBreakers - one for the input blocks and one for the operation itself.
+         * The input blocks don't count against the memory usage for the limited operator that we
+         * build.
+         */
+        List<Page> input = CannedSourceOperator.collectPages(simpleInput(driverContext().blockFactory(), between(1_000, 10_000)));
```

Comment:
We want this blockFactory to never circuit break, right? Just do the accountancy.

Reply:
Yeah. Right now we actually don't assert that we release all the blocks either. I'll grab that in a follow up. The trouble, I think, is just a missing `try` block around processing the page. I *think* it's that simple, but I want to dig more into the failure cases in a follow up.